### PR TITLE
Rolling schedule#109

### LINKED ISSUE #109: Schedule auto-updates
The schedule shows seven days in advance. At midnight each night, waterings are automatically added to the schedule in keeping with my plants.times_per_week.

### PR DESCRIPTION
**Pull request by bendelonlee**
Heroku scheduler should call rake generate_waterings.
This rake task calls Scheduler.generate_waterings_for_a_week_from_today
The scheduler PORO generates waterings for that day for all plants based on their thirstiness.
It calls a newly created instance method on the Plant model: #projected_thirstiness_for(date)
All newly created methods are 100% covered, overall coverage at 94%.

closes #109

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -8,3 +8,7 @@ task :rainy_day => :environment do
   RainyDayJob.set.perform_now(:just_set_up)
   puts "let the texting begin"
 end
+
+task :generate_waterings => :environment do
+  Scheduler.generate_waterings(:one_week_from_today)
```
Comment (bendelonlee) [inline Rakefile]: fix this.

File: lib/modules/our_array_methods.rb [hunk 2]
```diff
@@ -10,4 +10,9 @@ def self.spread_evenly(length, step, offset = 0)
     end
     arr
   end
+  def self.next(array, step)
+    expected = step * (array.length + 1)
+    actual_so_far = array.sum
+    (expected - actual_so_far).to_i
+  end
```
Comment (bendelonlee) [inline lib/modules/our_array_methods.rb]: delete

File: Rakefile [hunk 3]
```diff
@@ -8,3 +8,8 @@ task :rainy_day => :environment do
   RainyDayJob.set.perform_now(:just_set_up)
   puts "let the texting begin"
 end
+
+task :generate_waterings => :environment do
+  Scheduler.generate_waterings_for_a_week_from_today
```
Comment (le3ah) [inline Rakefile]: Super cool that Heroku can call our rake tasks

File: spec/features/users/user_generates_schedule_when_adding_a_plant_spec.rb [hunk 4]
```diff
@@ -67,7 +67,7 @@
 
     visit schedules_path
 
-    within("div[name='#{Date.today.strftime('%b%d')}']") do
+    within("div[name='#{Time.now.to_date.strftime('%b%d')}']") do
       expect(page).to have_content(plant.name, count: 1)
```
Comment (le3ah) [inline spec/features/users/user_generates_schedule_when_adding_a_plant_spec.rb]: Great testing specificity

File: spec/models/scheduler_spec.rb [hunk 5]
```diff
@@ -20,4 +20,32 @@
     expect(Date.parse(plant_2.waterings.third.water_time.to_s)).to eq(Date.parse(5.days.from_now.localtime.to_s))
     expect(Date.parse(plant_2.waterings.fourth.water_time.to_s)).to eq(Date.parse(7.days.from_now.localtime.to_s))
   end
+
+  it '.generate_waterings_for_a_week_from_today' do
+    def a_day_goes_by(plants)
+      plants.each do |plant|
+        plant.waterings.each do |watering|
+          #similating the passage of time
```
Comment (le3ah) [inline spec/models/scheduler_spec.rb]: Not sure if we want to leave this commented out code in there, but if we do, I think it's "simulating" 🤓

File: app/models/scheduler.rb [hunk 6]
```diff
@@ -8,9 +8,21 @@ def self.generate_plant_schedule(plant)
     end
   end
 
+  def self.generate_waterings_for_a_week_from_today
+    a_week_from_today = Time.now.to_date + 7.days
```
Comment (asmolentzov) [inline app/models/scheduler.rb]: This is great!

File: Rakefile (hunk 3 above)
Comment (asmolentzov) [reply Rakefile]: 🌟

**Pull request merged**